### PR TITLE
[Communication]: Android: Regenerate Chat clients with swagger for Preview3

### PR DESCRIPTION
**Pull request by JianpingChen**

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java [hunk 1]
```diff
@@ -57,21 +87,21 @@ public void sendChatReadReceipt(String chatThreadId, SendReadReceiptRequest send
...
+     * 
      * @param chatThreadId The thread id to send the message to.
      * @param sendChatMessageRequest Details of the message to send.
      * @param callback the Callback that receives the response.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      */
-    public void sendChatMessage(String chatThreadId, SendChatMessageRequest sendChatMessageRequest, final Callback<String> callback) {
+    public void sendChatMessage(String chatThreadId, SendChatMessageRequest sendChatMessageRequest, final Callback<SendChatMessageResult> callback) {
```
Comment (marche0133) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:98]: As discussed, we want this callback return type to be String, which is the message id of the SendChatMessageResult. Not sure if you unpack the callback here to String return type without touching Implement file.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java [hunk 2]
```diff
@@ -84,21 +102,21 @@
...
      * @param chatThreadId The thread id to send the message to.
      * @param sendChatMessageRequest Details of the message to send.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return the ID of the message sent.
+     * @return result of the send message operation.
      */
-    public Response<String> sendChatMessageWithRestResponse(String chatThreadId, SendChatMessageRequest sendChatMessageRequest) {
+    public Response<SendChatMessageResult> sendChatMessageWithRestResponse(String chatThreadId, SendChatMessageRequest sendChatMessageRequest) {
```
Comment (marche0133) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java:113]: We should add simple convert here to change Response<SendChatMessageResult> to Response<String> which is the message id. Remember also update JavaDoc

**Review by marche0133 — COMMENTED** [review]
In convienence layer, in chatThreadClient and chatThreadAsyncClient we need to add a method to add single participant 

Reference:
8. Add convenience method to add 1 member

Currently the SDKs allow the creation of members in batch. The suggestion is to add a convenience method to allow the addition of just one member. Add support for AddMember on the sdks

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java (hunk 1 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java]: I made some manual change to convenience layer for that purpose. Please take a look.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java (hunk 2 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java]: Made manual change to convenience layer.

**Review by marche0133 — APPROVED** [review]

**Pull request merged**